src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,199 @@
+
+
+			// we have to set icon ("This is the only user-visible content that's required", says doc).
+
+			// method 1
+			string packageName = Android.App.Application.Context.PackageName;
+			Android.Content.PM.ApplicationInfo info = null;
+			try
+			{
+				info = Android.App.Application.Context.ApplicationContext.PackageManager.GetApplicationInfo(packageName, 0);
+			}
+			catch
+			{
+			}
+
+			int iconId;
+			if (info != null)
+			{
+				iconId = info.Icon;
+			}
+			else
+			{
+				// try another method - using more resources (CPU and memory), as Intent is involved
+				Android.Content.Intent intent = Android.App.Application.Context.PackageManager.GetLaunchIntentForPackage(Android.App.Application.Context.PackageName);
+				Android.Content.PM.ResolveInfo resolveInfo = Android.App.Application.Context.PackageManager.ResolveActivity(intent, Android.Content.PM.PackageInfoFlags.MatchDefaultOnly);
+				iconId = resolveInfo.IconResource;
+			}
+
+			builder.SetSmallIcon(iconId);
+
+
+			// for somewhat older Android, we need to set Priority (deprecated in API26)
+
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.O)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder.SetPriority(0); // PRIORITY_DEFAULT
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+
+			// mirroring / bridging of notification
+			builder.SetLocalOnly((notification.NotificationMirroring == NotificationMirroring.Disabled));
+
+			// expiration time
+			if((notification.ExpirationTime != null) && (notification.ExpirationTime.HasValue) )
+			{
+				int seconds = (int)(notification.ExpirationTime.Value - DateTime.Now).TotalSeconds;
+				if(seconds>0)
+				{
+					builder.SetTimeoutAfter(seconds * 1000);
+				}
+
+			}
+
+
+			// action - opening app from tap on Toast (App.OnActivated)
+
+			var intentAction = Android.App.Application.Context.PackageManager.GetLaunchIntentForPackage(packageName);
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var xmlDoc = new XmlDocument();
+			try
+			{
+				xmlDoc.LoadXml(notification.Content.GetXml());  // Windows.Data.Xml.Dom is almost not implemented, so we have to use System.Xml
+				var childToast = xmlDoc.GetElementsByTagName("toast");
+				if(childToast.Count == 1)
+				{
+					var childLaunch = childToast.Item(0).Attributes.GetNamedItem("launch");
+					if(childLaunch != null)
+					{
+						toastArgument = childLaunch.Value;
+					}
+				}
+			}
+			catch
+			{
+				// error parsing XML - maybe simple not 'launch' attribute

```

Comment:
Log an error, avoid using exceptions as the default flow (e.g. test for item length, then launch presence)

Reply:
This is not error condition. Using 'launch' attribute is optional. If it is defined, it should be used, but if not defined, then argument is simply empty.

Reply:
In this case, if may be a parsing error, which needs to be logged. If the `launch` condition is optional do not use the exception flow to view a non-exceptional flow.

Reply:
I don't know how to test for existence object on this path: `childToast.Item(0).Attributes.GetNamedItem("launch");`. So, I used try/catch syntax.